js/forms.js

```diff
@@ -279,85 +279,181 @@
      * Auto complete plugin  *
      *************************/
     $.fn.autocomplete = function (options) {
-      // Defaults
       var defaults = {
-        data: {}
-      };
+        data: {},
+        maxElementsAmount: null
+      },
+      currentLi = 0,
+      autocompleteOption = null,
+      currentAutocompleteVal = '';
 
       options = $.extend(defaults, options);
 
-      return this.each(function() {
-        var $input = $(this);
-        var data = options.data,
-            $inputDiv = $input.closest('.input-field'); // Div to append on
+      // Set input value
+      function setIputValue(ulElement, inputElement) {
+        ulElement.on('click', 'li', function () {
+          inputElement.val($(this).text().trim());
+          ulElement.empty();
+        });
+      }
+
+      // highlight elements
+      function highlight(string, $el) {
+        var img = $el.find('img');
+        var matchStart = $el.text().toLowerCase().indexOf("" + string.toLowerCase() + ""),
+        matchEnd = matchStart + string.length - 1,
+        beforeMatch = $el.text().slice(0, matchStart),
+        matchText = $el.text().slice(matchStart, matchEnd + 1),
+        afterMatch = $el.text().slice(matchEnd + 1);
+        $el.html("<span>" + beforeMatch + "<span class='highlight'>" + matchText + "</span>" + afterMatch + "</span>");
+        if (img.length) {
+          $el.prepend(img);
+        }
+      }
+
+      // function on keyUp
+      function keyUp(e, data, $input, $autocomplete) {
+
+        // select element on Enter
+        if (e.which === 13) {
+          $autocomplete.find('li')[currentLi].click();
+          return;
+        }
+
+        // hide list on ESC button
+        if (e.which === 27) {
+          $input.blur();
+        }
+
+        // prevent refreshing list on pressing ctrl, shif, alt, arrow, f1-f12, etc
+        // helpful if we selected something using arrow and we press ctrl for example
+        if(
+        [9, 16, 17, 18, 19, 20, 33, 34, 35, 36, 37, 38, 39, 40, 45, 112, 113, 114, 115, 116, 117, 118, 119, 120, 121, 122, 123].indexOf(e.which) < 0
```

Comment:
I'm not so sure about that, but if you could provide solution like that and it will work we can use that.